aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -108,66 +235,30 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_backward_cuda(const Tensor& grad_o
 }
 
 std::tuple<Tensor, Tensor> batch_norm_stats_cuda(const Tensor& self, double epsilon) {
-  bool use_channels_last_kernel = batch_norm_use_channels_last_kernels(self);
+  const int64_t n_input = self.size(1);
+  auto options = self.options().dtype(
+      at::toAccumulateType(self.scalar_type(), /*is_cuda=*/true));
+  auto save_mean = at::empty({n_input}, options);
+  auto save_var = at::empty({n_input}, options);
 
-  return AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, self.scalar_type(), "batch_norm_stats_cuda", [&] {
-    if (cuda::detail::canUse32BitIndexMath(self)) {
-      if (use_channels_last_kernel) {
-        return batch_norm_stats_channels_last_cuda_template<scalar_t>(self, epsilon);
-      } else {
-        return batch_norm_stats_cuda_template<scalar_t, int32_t>(self, epsilon);
-      }
-    } else {
-      return batch_norm_stats_cuda_template<scalar_t, int64_t>(self, epsilon);
-    }
-  });
+  // FIXME: Epsilon parameter isn't required, we don't take the reciprocal
+  batch_norm_var_mean(self, save_var, save_mean);
+  return std::tuple<Tensor, Tensor>(save_mean, save_var);
 }
 
 Tensor batch_norm_elemt_cuda(const Tensor& self, const c10::optional<Tensor>& weight_opt, const c10::optional<Tensor>& bias_opt,
                              const Tensor& mean, const Tensor& invstd, double epsilon) {
-  // See [Note: hacky wrapper removal for optional tensor]
-  c10::MaybeOwned<Tensor> weight_maybe_owned = at::borrow_from_optional_tensor(weight_opt);
-  const Tensor& weight = *weight_maybe_owned;
-  const Tensor& bias = c10::value_or_else(bias_opt, [] {return Tensor();});
-
   auto output = at::empty_like(self, self.suggest_memory_format());
-  at::native::batch_norm_elemt_cuda_out(self, weight, bias, mean, invstd, epsilon, output);
+  // FIXME: Epsilon parameter isn't required, we don't take the reciprocal
+  batch_norm_elementwise(output, self, weight_opt, bias_opt, mean, invstd);
```

Comment:
replacing batch_norm_elemt with TI kernel as you have here is probably beneficial, but needs benchmarking.